protos/local/activity_result.proto

```diff
@@ -5,27 +5,33 @@ package coresdk.activity_result;
 import "common.proto";
 import "temporal/api/failure/v1/message.proto";
 
-/// Used to report activity completion to core and to resolve the activity in a workflow activation
+/**
+ * Used to report activity completion to core and to resolve the activity in a workflow activation
+ */
 message ActivityResult {
     oneof status {
         Success completed = 1;
         Failure failed = 2;
-        Cancelation canceled = 3;
+        Cancellation cancelled = 3;
     }
 }
 
-/// Used in ActivityResult to report cancellation
-message Cancelation {
-    common.Payload details = 1;
-}
-
-/// Used in ActivityResult to report successful completion
+/** Used in ActivityResult to report successful completion */
 message Success {
     common.Payload result = 1;
 }
 
-/// Used in ActivityResult to report failure
+/** Used in ActivityResult to report failure */
 message Failure {
     temporal.api.failure.v1.Failure failure = 1;
 }
 
+/**
+ * Used in ActivityResult to report cancellation from both Core and Lang.
+ * When Lang reports a cancelled ActivityResult, it must put a CancelledFailure in the failure field.
```

Comment:
Wait, shouldn't this be in the `cancelled` field?

Reply:
Oh, you mean in the field of *this* struct. Maybe add a word or two to clarify that